contracts/smart-contract-wallet/modules/SessionKeyModule.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {BaseAuthorizationModule, UserOperation, ISignatureValidator} from "./BaseAuthorizationModule.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@account-abstraction/contracts/core/Helpers.sol";
+import "./SessionValidationModules/ISessionValidationModule.sol";
+
+struct SessionStorage {
+    bytes32 merkleRoot;
+}
+
+contract SessionKeyManager is BaseAuthorizationModule {
+    /*
+     * @dev mapping of Smart Account to a SessionStorage
+     * Info about session keys is stored as root of the merkle tree built over the session keys
+     */
+    mapping(address => SessionStorage) internal userSessions;
```

Comment:
would we need to retrieve merkle root in the backend service? 
asking if need to make this public or there is a read method, if applicable

Reply:
ok, resolved.